File: FizzBuzz/kotlin/src/main/kotlin/FiltersProvider.kt
object FiltersProvider {
	private var filters: MutableList<(Int) -> String> = mutableListOf()

	fun add(filter: (Int) -> String)
	{
		filters.add(filter)
	}

	fun removeAll()
	{
		filters.clear()
	}

	fun all(): MutableList<(Int) -> String>
	{
		return filters.toMutableList()
	}
}

File: FizzBuzz/Scala/src/main/scala/FiltersProvider.scala
object FiltersProvider {
  private var filters: List[(Int) => String] = Nil

  def all(): List[(Int) => String] = {
    return filters
  }

  def add(filter: (Int) => String) = {
    filters = filter::filters
  }

  def removeAll() {
    filters = Nil
  }
}
